Frameworks/BlockchainCertificates/Issuer.swift

```diff
@@ -287,3 +269,93 @@ public func ==(lhs: KeyRotation, rhs: KeyRotation) -> Bool {
 public func <(lhs: KeyRotation, rhs: KeyRotation) -> Bool {
     return lhs.on < rhs.on
 }
+
+func parseKeysV1(from dictionary: [String: Any], with keyName: String) throws -> [KeyRotation] {
+    guard let keyProperty = dictionary[keyName] else {
+        throw IssuerError.missing(property: keyName)
+    }
+    guard let keyData = keyProperty as? [[String: String]] else {
+        throw IssuerError.invalid(property: keyName)
+    }
+    
+    let parsedKeys = try keyData.enumerated().map { (index: Int, dictionary: [String : String]) throws -> KeyRotation in
+        do {
+            let rotation = try keyRotationSchedule(from: dictionary)
+            return rotation
+        } catch IssuerError.missing(let prop) {
+            throw IssuerError.missing(property: ".\(keyName).\(index).\(prop)")
+        } catch IssuerError.invalid(let prop) {
+            throw IssuerError.invalid(property: ".\(keyName).\(index).\(prop)")
+        }
+    }
+    
+    return parsedKeys
+}
+
+public func parseKeysV2(from dictionary: [String: Any], with keyName: String) throws -> [KeyRotation] {
+    guard let keyProperty = dictionary[keyName] else {
+        throw IssuerError.missing(property: keyName)
+    }
+    guard let keyData = keyProperty as? [[String: String]] else {
+        throw IssuerError.invalid(property: keyName)
+    }
+    
+    let parsedKeys = try keyData.enumerated().map { (index: Int, dictionary: [String : String]) throws -> KeyRotation in
+        do {
+            let rotation = try keyRotationScheduleV2(from: dictionary)
```

Comment:
fixed to pass keyRotationSchedule/keyRotationScheduleV2 as a function argument to single (unified) parseKeys function